src/JigsawStructureDataPacket.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+/*
+ * This file is part of BedrockProtocol.
+ * Copyright (C) 2014-2022 PocketMine Team <https://github.com/pmmp/BedrockProtocol>
+ *
+ * BedrockProtocol is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ */
+
+declare(strict_types=1);
+
+namespace pocketmine\network\mcpe\protocol;
+
+use pocketmine\network\mcpe\protocol\serializer\PacketSerializer;
+use pocketmine\network\mcpe\protocol\types\CacheableNbt;
+
+class JigsawStructureDataPacket extends DataPacket implements ClientboundPacket{
+	public const NETWORK_ID = ProtocolInfo::JIGSAW_STRUCTURE_DATA_PACKET;
+
+	/** @phpstan-var CacheableNbt<\pocketmine\nbt\tag\CompoundTag> */
+	private CacheableNbt $nbt;
+
+	/**
+	 * @generate-create-func
+	 * @phpstan-param CacheableNbt<\pocketmine\nbt\tag\CompoundTag> $nbt
+	 */
+	public static function create(CacheableNbt $nbt) : self{
+		$result = new self;
+		$result->nbt = $nbt;
+		return $result;
+	}
+
+	/** @phpstan-return CacheableNbt<\pocketmine\nbt\tag\CompoundTag> */
```

Comment:
same

Reply:
It's like this in all packets except for the StartGamePacket, so the majority of packets use the long form

Reply:
I didn't know about it, all right.